src/Service/PaymentStateDataService.php

```diff
@@ -130,7 +132,15 @@ public function getPaymentMethodType(string $contextToken): ?string
             return OneClickPaymentMethodHandler::getPaymentMethodCode();
         }
 
-        return $stateDataArray['paymentMethod']['type'];
+        $type = $stateDataArray['paymentMethod']['type'];
```

Comment:
Yes, sounds like it should be possible to use the payment method handler instead of `$stateDataArray['paymentMethod']['type']`. We would have to check for the stored payment methods and bcmc handlers to add the exceptions. What do you think @peterojo?

Reply:
I think with this approach the exceptions won't be needed as bcmc and stored PMs would be checked against the handler in the frontend

Reply:
I think it should work, we already maintain a mapping in `adyenConfiguration.paymentMethodTypeHandlers` that we would need to convert the `handler_adyen_...` -style formatted identifiers into the short method names